pages/timer/[id].js

```diff
@@ -0,0 +1,85 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+import LayoutSytle from "../../components/LayoutStyle";
+
+export default function Timer({ tasks }) {
+  const router = useRouter();
+  const { id } = router.query;
+  const [durationClock, setDurationClock] = useState(null);
+
+  useEffect(() => {
+    setDurationClock({ sec: duration, start: true });
+  }, []);
+
+  useEffect(() => {
+    let timer;
+    if (durationClock !== null) {
+      timer = setInterval(() => {
+        if (durationClock.sec > 0) {
+          setDurationClock((pre) => ({ sec: pre.sec - 1, start: pre.start }));
+        }
+      }, 1000);
+    }
+
+    return () => {
+      clearInterval(timer);
+    };
+  }, [durationClock]);
+
+  const task = tasks.find((task) => task.id === id);
+
+  if (durationClock === null || task === undefined) return;
+  const { taskName, duration } = task;
+
+  function displayDuration() {
+    const hour =
+      Math.floor(durationClock.sec / 3600) < 10
+        ? "0" + Math.floor(durationClock.sec / 3600)
+        : Math.floor(durationClock.sec / 3600);
+    const minutes =
+      durationClock.sec % 3600 === 0
+        ? "00"
+        : Math.floor(durationClock.sec / 3600) >= 1
+        ? Math.floor((durationClock.sec % 3600) / 60)
+        : Math.floor(durationClock.sec / 60) < 10
```

Comment:
hm, I think nested terniaries are not the best idea. I would recommend switching to something else, like if else or switch